tests/integration/buildcmd/test_build_terraform_applications.py

```diff
@@ -84,3 +128,137 @@ def test_exit_success_no_beta_features_flags_supplied_hooks(self):
         _, stderr, return_code = self.run_command(cmdlist)
         self.assertEqual(return_code, 0)
         self.assertEqual(stderr.strip().decode("utf-8"), "Terraform Support beta feature is not enabled.")
+
+
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+class TestBuildTerraformApplicationsWithZipBasedLambdaFunctionAndLocalBackend(BuildTerraformApplicationIntegBase):
+    terraform_application = Path("terraform/zip_based_lambda_functions_local_backend")
+    functions = [
+        "aws_lambda_function.from_localfile",
+        "aws_lambda_function.from_s3",
+        "module.level1_lambda.aws_lambda_function.this",
+        "module.level1_lambda.module.level2_lambda.aws_lambda_function.this",
+        "my_function_from_localfile",
+        "my_function_from_s3",
+        "my_level1_lambda",
+        "my_level2_lambda",
+    ]
+
+    @parameterized.expand(functions)
+    def test_build_and_invoke_lambda_functions(self, function_identifier):
+        build_cmd_list = self.get_command_list(
+            beta_features=True, hook_package_id="terraform", function_identifier=function_identifier
+        )
+        LOG.info("command list: %s", build_cmd_list)
+        _, stderr, return_code = self.run_command(build_cmd_list)
+        LOG.info(stderr)
+        self.assertEqual(return_code, 0)
+
+        self._verify_invoke_built_function(
```

Comment:
why not to call `sam local start-lambda --hook-package-id Terraform` instead of calling the normal sam local start-lambda